### ISSUE TITLE
COPY/PAST is not working for applications supported XCLIP

### ISSUE THREAD

**Issue by Golyshkin** [shmuz/far2m#53]
Actually FAR2M uses TTY Clipboard and COPY/PAST working fine in/from console and other many applications, but it's not working for some other applications for example Telegram

1. FAR2M <-> Telegram = NOK
2. TELEGRAM <-> NEDIT <-> FAR2M = OK

Is it possible to investigate how it can be fixed?

Reference: https://github.com/elfmz/far2l/issues/810?ysclid=lsljpjbldn114239135

**Comment by shmuz** [comment]
1. Are you running far2m in some TTY mode (e.g. TTY|Xi) or in "GUI" mode?
2. Does far2m behave differently from far2l in this aspect?

**Comment by Golyshkin** [comment]
> 1. Are you running far2m in some TTY mode (e.g. TTY|Xi) or in "GUI" mode?

I am running FAR2M using following created desktop file

> [Desktop Entry]
> Type=Application
> Name=far2m
> GenericName=far2m
> Comment=File and archive manager
> Exec=/usr/local/bin/far2m
> TryExec=/usr/local/bin/far2m
> Terminal=false
> Categories=Utility;FileManager;
> Icon=far2m
> StartupNotify=true

> 2. Does far2m behave differently from far2l in this aspect?

There is no such info, since I am not using FAR2L anymore...

**Comment by shmuz** [comment]
OK then I assume you're running it in GUI mode. This is the mode I use 95% of the time.
It can be verified by the command `lua:=actl.WinPortBackend()`

Here is what I did to test this issue:
1. Installed telegram-desktop on Ubuntu 22.04
2. Tried copy-paste between far2m and telegram and it worked for me in both direction.

I'm somewhat embarrassed that I can't repeat the errors you report.

**Comment by shmuz** [comment]
Please tell if you build far2m from sources or get it some other way.

**Comment by Golyshkin** [comment]
Below are answers on your questions above:
![image](https://github.com/shmuz/far2m/assets/772340/baffb000-c73a-496d-a0d9-d6483a31d43f)
![image](https://github.com/shmuz/far2m/assets/772340/c89e65ef-14ee-4cfb-99a6-fffeb5793f2f)

If you have any additional questions, please feel free to ask me...

**Comment by Golyshkin** [comment]
> Please tell if you build far2m from sources or get it some other way.

Yes, I am using manual compiled FAR2M version and updating it once per week.

**Comment by shmuz** [comment]
I'm trying to understand what is different between our systems.
As I'm doing the tests on the same OS as you have, I though that maybe your far2m was different from mine because you (for example) installed it from PPA. Now that I know you build it from sources I'm again out of ideas.

**Comment by akruphi** [comment]
>  1. FAR2M <-> Telegram = NOK

I had such problem Far2l <-> Telegram in Wayland: https://github.com/elfmz/far2l/issues/1350#issuecomment-1443347317

You can check
```
echo $XDG_SESSION_TYPE
echo $GDK_BACKEND
echo $DESKTOP_SESSION
```
and if you under Wayland you can try running from console: `GDK_BACKEND=x11 far2m`

**Comment by shmuz** [comment]
@akruphi
Here is what I see (and copy-paste between far2m and telegram works in both directions).
```
wayland

ubuntu
```

**Comment by Golyshkin** [comment]
> > 1. FAR2M <-> Telegram = NOK
> 
> I had such problem Far2l <-> Telegram in Wayland: [elfmz#1350 (comment)](https://github.com/elfmz/far2l/issues/1350#issuecomment-1443347317)
> 
> You can check
> 
> ```
> echo $XDG_SESSION_TYPE
> echo $GDK_BACKEND
> echo $DESKTOP_SESSION
> ```
> 
> and if you under Wayland you can try running from console: `GDK_BACKEND=x11 far2m`

Yes, you right, this issue is not reproducible for 

`GDK_BACKEND=x11 far2m`

but re-opens the  https://github.com/shmuz/far2m/issues/49

**Comment by Golyshkin** [comment]
According to suggestion to check system ENV variables for FAR2M

> echo $XDG_SESSION_TYPE
> echo $GDK_BACKEND
> echo $DESKTOP_SESSION

![image](https://github.com/shmuz/far2m/assets/772340/399b2676-eb92-4b1b-bbc9-dc374119323b)

**Comment by akruphi** [comment]
Wayland is new fast developing technology and its implementation often differs significantly in different systems and in different versions of the same system. If it possible please collect info with details (OS version, Wayland version, desktop environment version etc.) with and without bug and add issue about it in https://github.com/elfmz/far2l/issues/1350 or in far2l new issue.

**Comment by Golyshkin** [comment]
> Wayland is new fast developing technology and its implementation often differs significantly in different systems and in different versions of the same system. If it possible please collect info with details (OS version, Wayland version, desktop environment version etc.) for cases with and without bug and add issue about it in [elfmz#1350](https://github.com/elfmz/far2l/issues/1350) or in far2l new issue.

Please provide which commands output are u interested to see in the [elfmz#1350](https://github.com/elfmz/far2l/issues/1350)? I'll do that.

**Comment by akruphi** [comment]
Actual issues about wayland problems: https://github.com/elfmz/far2l/issues/2041 and https://github.com/elfmz/far2l/issues/2040

**Comment by unxed** [comment]
Suggesting to switch Wayland off until upstream bugs causing UX problems are fixed

https://github.com/elfmz/far2l/pull/2064

**Comment by shmuz** [comment]
@Golyshkin 
Please try now.

**Comment by Golyshkin** [comment]
2.6.0-2024-03-18-e05501c3-beta fixes this issue. thnx
